compiler/AST/iterator.cpp

```diff
@@ -412,14 +456,16 @@ buildZip1(IteratorInfo* ii, Vec<BaseAST*>& asts, BlockStmt* singleLoop) {
   }
 
   if (WhileStmt* stmt = toWhileStmt(singleLoop)) {
+    // By the time we get here, the condExpr has been passed through _cond_test
+    // and the result stored in a temp, so condExprForTmpVariableGet just
+    // returns the temp result.  So we can simply say:  (.= this more tmp)
+    // This assumes the backend will implicitly cast tmp to the type of more
```

Comment:
I don't disagree that the c backend will implicitly coerce bools to ints. And there shouldn't be any backend that complain about it (some do complain about int to bool casts.)

That said I think `condExpr` should at least be cast to an before assigning it to `more`

The chapel code: 

```chapel
  var b = true;
  proc foo (i: int) { writeln(i); }
  foo(b);

```

generates the following c code (at the call site): 

```c
  int64_t call_tmp_chpl;
  call_tmp_chpl = ((int64_t)(true));
  foo_chpl(call_tmp_chpl);
```

We are careful to insert explicit casts into the AST even for coercians that are implicit at the chapel level, and I think it's better to maintain that